doc/Uplink.xml

```diff
@@ -172,15 +174,56 @@
     <title>Uplink</title>
     <section>
       <title>Connection Establishment</title>
-      <para>The device initiates the connection to the cloud service. <xref linkend="_Ref505091745" /> shows the three phases. In the first phase the device acts as TLS client that connects to the cloud service. Please note that the figure only shows the most relevant packets. Details of the TCP and TLS exchange are out of scope of this specification. The second phase includes the connection upgrade to HTTP/2 which is confirmed by the cloud service with a 101 HTTP response. The third phase of the connection than fully complies to an HTTP2 connection as if it were initiated from the cloud service.</para>
-      <figure xml:id="_Ref505091745">
-        <title>Connection initiation from the device</title>
-        <mediaobject>
-          <imageobject>
-            <imagedata fileref="media/Uplink/image4.svg" contentwidth="128mm" />
-          </imageobject>
-        </mediaobject>
-      </figure>
+      <para>There are two connection mechanism available for a device to connect to a remote client. The device can either use an http/2 reverse tunnel or a WebSocket.</para>
+      <section>
+        <title>Establishing the reverse tunnel</title>
+        <para>
+          The device initiates the connection to the cloud service. This is done in three phases.
+          Phase 2 and 3 depend on the protocol being used and are detailed below.
+          In the first phase the device acts as TLS client that connects to the cloud service.
+          Details of the TCP and TLS exchange are out of scope of this specification.
+        </para>
+        <section>
+          <title>Native h2c-reverse</title>
+          <para> <!-- HTTP/2 normal -->
+            The second phase includes the connection upgrade to HTTP/2 h2c-reverse protocol which is confirmed by the cloud service with a 101 HTTP response.
+            The third phase of the connection than fully complies to an HTTP2 connection as if it were initiated from the cloud service.
+            Please note that the figure only shows the most relevant packets.
+          </para>
+          <figure xml:id="_Ref505091745">
+            <title>Connection initiation from the device</title>
+            <mediaobject>
+              <imageobject>
+                <imagedata fileref="media/Uplink/image4.svg" contentwidth="128mm" />
+              </imageobject>
+            </mediaobject>
+          </figure>
+        </section>
+        <section>
+          <title>WebSocket h2c-reverse</title>
+          <para> <!-- WebSocket normal -->
+            The second phase includes the connection upgrade to a WebSocket with the reverse HTTP2 subprotocol which is confirmed by the cloud service with a 101 HTTP response.
+            The third phase of the connection then fully complies to an HTTP2 connection as if it were initiated from the cloud services, using WebSocket binary frames.
```

Comment:
Suggesting correction;

`with the reverse HTTP/2 sub-protocol`

Also suggesting


`.. fully complies with a HTTP/2 connection ...`